azavea/opendataphilly-ckan
# Update Ansible deployment to work in production environment

## Overview

* Remove S3 database backups (they weren't working, and we have RDS snapshots)
* Add temp Nginx config for LetsEncrypt verification 
* Refactor `group_vars` for 2.8 deployment

It looks like the LetsEncrypt Ansible task was only tested in the context of an OpenDataPhilly EC2 instance that is already provisioned and has an existing SSL certificate. The problem is that, when the `ckan-odp-configuration` role reaches this task on a fresh EC2 instance, there is no SSL certificate, Nginx only listens on `443` and will fail to start because the SSL certificate is missing, and LetsEncrypt is configured to store the verification challenge behind Nginx.

The "Hello World" Apache endpoint is still running on `80`, but we can't use it for renewals. The Apache service restarts and the web root becomes inaccessible once the playbook finishes executing. The solution I came up with was to offer a temporary Nginx configuration that will listen on `80` for the initial certificate verification, and gets overwritten once LetsEncrypt finishes.

Connects https://github.com/azavea/urban-apps/issues/164

## Testing Instructions

Klaas and I tested this on an EC2 instance when we were attempting a dry run of the upgrade to CKAN 2.8.

Follow the [Deployment](
https://github.com/azavea/opendataphilly-ckan#deployment) instructions in the README, and ensure the Ansible playbook finishes.
